### PR TITLE
adds assert messages to std.concurrency

### PR DESCRIPTION
**Pull request by burner**

### CODE REVIEW

File: std/concurrency.d [hunk 1]
```diff
@@ -186,7 +186,8 @@ private
     {
         foreach (i, t1; T)
         {
-            static assert(isFunctionPointer!t1 || isDelegate!t1);
+            static assert(isFunctionPointer!t1 || isDelegate!t1,
+                    "Not all passed T are function pointer or delegates");
```
Comment (thewilsonator) [inline std/concurrency.d]: function pointer_s_. Also you have the index of the T that is not a function pointer or delegate, you may as well give that as part of the message.

File: std/concurrency.d [hunk 2]
```diff
@@ -198,7 +199,8 @@ private
 
                 foreach (t2; T[i + 1 .. $])
                 {
-                    static assert(isFunctionPointer!t2 || isDelegate!t2);
+                    static assert(isFunctionPointer!t2 || isDelegate!t2,
+                            "Not all passed T are function pointer or delegates");
```
Comment (thewilsonator) [inline std/concurrency.d]: this static assert is redundant with the one above, no?

File: std/concurrency.d (hunk 1 above)
Comment (burner) [reply std/concurrency.d]: done

File: std/concurrency.d (hunk 2 above)
Comment (burner) [reply std/concurrency.d]: done

**Comment by burner** [comment]
@thewilsonator thanks

**Pull request merged**